client/asset/dcr/dcr.go

```diff
@@ -353,37 +381,36 @@ func (dcr *ExchangeWallet) Fund(value uint64, nfo *dex.Asset) (asset.Coins, erro
 
 out:
 	for {
-		// On each loop, find the smallest UTXO that is enough for the value. If
-		// no UTXO is large enough, add the largest and continue.
+		// On each loop, find the smallest UTXO that is enough. If no UTXO is large
+		// enough, add the largest and continue.
 		var txout *compositeUTXO
 		for _, txout = range utxos {
```

Comment:
If `addUTXO` is used above, we have enough funds. The `for` loop `break`s and we don't care about `utxos` anymore. If `addUTXO` wasn't called, nothing has been added yet, so I add the largest utxo, which I already know is not big enough, and loop again. In the latter case, it has to be removed to prevent adding it twice.

Reply:
Ah, I missed the `out` in `break out`.  Thanks for walking me through that.